src/shared/nt-results/ResultsByLoad.react.js

```diff
@@ -144,48 +169,73 @@ class ResultsByLoad extends React.Component {
+      .curve(d3.curveCardinal.tension(0))
+
+    return this.state.paths.map((path, index) => (
+      <path
+        key={`path-${index}`}
+        className={styles.nt__path}
+        d={line(path)}
+        strokeLinecap='round'
+        fill='none'
+      />
+    ))
+  }
+
   renderLinks = () => this.state.links.map((link, index) => {
+    const selected = this.isSelected(link.source.id) ||
+      this.isSelected(link.target.id)
+
+    const className = classNames(styles.nt__line, {
+      [styles['--selected']]: selected,
+      [styles['--not-selected']]: this.state.selectedNodeId && !selected,
+    })
+
     return (
       <g key={`link-${index}`}>
-        {this.renderGradient(`gradient-${index}`, link.source, link.target)}
+        {this.renderGradient(`gradient-${index}`, link)}
         <path
-          className={styles.line}
-          d={`M ${link.source.x}, ${link.source.y}
-            C ${link.source.x + 100},${link.source.y}
-              ${link.target.x - 100}, ${link.target.y}
-              ${link.target.x}, ${link.target.y}`}
+          className={className}
+          d={this.getCurveData(link)}
           markerEnd='url(#arrowHead)'
           stroke={`url(#gradient-${index})`}
         />
       </g>
     )
   })
 
-  renderNodes = () => this.state.nodes.map((node, index) => (
-    <g key={index} transform={`translate(${node.x || 0},${node.y || 0})`}>
-      <rect className={styles.node} x={-5} y={-8} width={10} height={16} />
-      <text x={-5} dy={25}>{node.type}</text>
-    </g>
-  ))
+  renderNodes = () => this.state.nodes.map((node, index) => {
+    const className = classNames(styles.nt__node, {
+      [styles['--not-selected']]: this.state.selectedNodeId &&
+        !this.isSelected(node.id)
+    })
+    const translate = `translate(${node.x || 0},${node.y || 0})`
+
+    return (
+      <g key={index} className={className} transform={translate}>
+        <rect
+          x={-5}
+          y={-8}
+          width={10}
+          height={16}
+          onMouseEnter={this.handleMouseEnterNode(node.id)}
+          onMouseLeave={this.handleMouseLeaveNode}
+        />
+        <text dy={25} textAnchor='middle'>{node.type}</text>
+      </g>
+    )
+  })
 
   render () {
     return (
       <svg width={this.props.width} height={this.props.height}>
-        <linearGradient id='gradient'>
-          <stop className={styles.from} offset='10%' />
-          <stop className={styles.to} offset='90%' />
-        </linearGradient>
-        <marker
-          id='arrowHead'
-          viewBox='0 -5 10 10'
-          refX='12'
-          orient='auto'
-          markerWidth='10'
-          markerHeight='10'
-        >
-          <path d='M 0,-5 L 8,0 L 0,5' className={styles.arrow} />
-        </marker>
-        {this.renderPaths()}
+        {this.renderLinearGradientRef()}
+        {this.renderArrowMarker()}
+        {!this.state.selectedNodeId && this.renderPaths()}
         {this.renderLinks()}
         {this.renderNodes()}
```

Comment:
Component didn't re-render after receiving props from graphql. I have to refresh result page or go to other page and come back again. 
- I guess you have to separate a component into pure component & container component. Then, a pure component will re-render when props change. 
- or, implement `componentWillReceiveProps` method